---
title: "SURVEY â€“ meta-analytic practices in neurobiology animal studies"
author: "Y. Yang, M. Lagisz , S. Nakagawa"
date: "July 2021"
output:
  html_document:
    toc: true
    toc_depth: 4
    number_sections: false
    toc_float:
      collapsed: false
      smooth_scroll: false
    fig.align: "center"
    fig_caption: true
    error: false
    warning: false
    message: false
    echo: false
    tidy: true
    cache: true
    df_print: paged
    code_folding: hide
editor_options: 
  chunk_output_type: console
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = F, warning = F, message = F)
pacman::p_load(magrittr, tidyr, stringr, ggplot2, cowplot, tidyverse, here, readxl)
#library(patchwork),  # for arranging ggplots in panels
library(devtools)
#devtools::install_github('Mikata-Project/ggthemr')
#devtools::install_github('cttobin/ggthemr')
library(ggthemr) #see https://github.com/Mikata-Project/ggthemr#palettes
# to remove all ggthemr effects later:
#ggthemr_reset()
ggthemr('fresh') #select one ggplot theme to be used
```



## Aims  
The main objective of the literature survey was to characterize key meta-analytic practices currently used in papers published in in the field of experimental animal studies in neurology. We aimed for a representative sample of 50-100 meta-analytic papers on non-human animals from the most relevant journals in the field.  

## Literature search  
We first identified top ranked 10% journals in "Neuroscience" Scopus subject area. This way, we collated a list of 115 journals alongside with their ISSN: 

#### Table S1  
Nr | Journal name |  ISSN   
---| -------------| -------------    
1.  | Nature Neuroscience  |   1546-1726   
2.  | Nature Reviews Neuroscience  | 1471-0048  
3.  | Nature Reviews Neurology  |   1759-4766  
4.  | Progress in Retinal and Eye Research  |   1873-1635  
5.  | Trends in Cognitive Sciences  |  1879-307X  
6.  | Neuron  | 1097-4199  
7.  | The Lancet Psychiatry  | 2215-0374  
8.  | Annual Review of Neuroscience  | 1545-4126  
9.  | Acta Neuropathologica  | 1432-0533  
10.  | Alzheimer's and Dementia  | 1552-5279  
11.  | Molecular Psychiatry  | 1476-5578  
12.  | Sleep Medicine Reviews  | 1532-2955  
13.  | Biological Psychiatry  | 1873-2402  
14.  | Progress in Neurobiology  | 1873-5118  
15.  | Molecular Neurodegeneration  | 1750-1326  
16.  | Trends in Neurosciences  | 1878-108X  
17.  | Ageing Research Reviews  | 1568-1637  
18.  | EMBO Journal  | 1460-2075  
19.  | Annals of Neurology  | 1531-8249  
20.  | Neuroscience and Biobehavioral Reviews  | 1873-7528  
21.  | Frontiers in Neuroendocrinology  | 1095-6808  
22.  | Movement Disorders  | 1531-8257  
23.  | Nature Human Behaviour  | 2397-3374  
24.  | Cellular and Molecular Life Sciences  | 1420-9071  
25.  | Brain  Behavior  and Immunity  | 1090-2139  
26.  | Current Biology  | 1879-0445  
27.  | Journal of Abnormal Psychology  | 1939-843X  
28.  | Current Opinion in Neurobiology  | 1873-6882  
29.  | Neuropathology and Applied Neurobiology  | 1365-2990  
30.  | Epilepsia  | 1528-1167  
31.  | Journal of Cerebral Blood Flow and Metabolism  | 1559-7016  
32.  | PLoS Biology  | 1545-7885  
33.  | Neural Networks  | 1879-2782  
34.  | Translational Neurodegeneration  | 2047-9158  
35.  | Neuroscientist  | 1089-4098  
36.  | GLIA  | 1098-1136  
37.  | Cell Death and Disease  | 2041-4889  
38.  | eLife  | 2050-084X  
39.  | NeuroImage  | 1095-9572  
40.  | Current Neuropharmacology  | 1875-6190  
41.  | Chronic Stress  | 2470-5470  
42.  | npj Parkinson's Disease  | 2373-8057  
43.  | Journal of Neuroinflammation  | 1742-2094  
44.  | Brain Stimulation  | 1876-4754  
45.  | Molecular Neurobiology  | 1559-1182  
46.  | Neurocomputing  | 0925-2312  
47.  | Molecular Autism  | 2040-2392  
48.  | Multiple Sclerosis Journal  | 1477-0970  
49.  | Neural Computation  | 1530-888X  
50.  | Alzheimer's Research and Therapy  | 1758-9193  
51.  | Brain Pathology  | 1750-3639  
52.  | Annals of the New York Academy of Sciences  | 1749-6632  
53.  | Cerebral Cortex  | 1460-2199  
54.  | Neurobiology of Stress  | 2352-2895  
55.  | Current Topics in Behavioral Neurosciences  | 1866-3389  
56.  | Biological Psychiatry: Cognitive Neuroscience and Neuroimaging  | 2451-9022  
57.  | Cognitive Computation  | 1866-9964  
58.  | Neurobiology of Disease  | 1095-953X  
59.  | Dialogues in Clinical Neuroscience  | 1958-5969  
60.  | Translational Psychiatry  | 2158-3188  
61.  | Current Neurology and Neuroscience Reports  | 1534-6293  
62.  | Human Brain Mapping  | 1097-0193  
63.  | Translational Stroke Research  | 8686-01X  
64.  | DMM Disease Models and Mechanisms  | 1754-8411  
65.  | Acta neuropathologica communications  | 2051-5960  
66.  | Journal of Pain  | 1528-8447  
67.  | Neurobiology of Aging  | 1558-1497  
68.  | Developmental Science  | 1467-7687  
69.  | Experimental Neurology  | 1090-2430  
70.  | Developmental Neurobiology  | 1932-846X  
71.  | Psychoneuroendocrinology  | 1873-3360  
72.  | IEEE Transactions on Neural Systems and Rehabilitation Engineering  | 1558-0210  
73.  | Journal of NeuroImmune Pharmacology  | 1557-1904  
74.  | Bipolar Disorders  | 1399-5618  
75.  | Developmental Medicine and Child Neurology  | 1469-8749  
76.  | Network Neuroscience  | 2472-1751  
77.  | British Journal of Ophthalmology  | 1468-2079  
78.  | Current Opinion in Behavioral Sciences  | 2352-1546  
79.  | PLoS Computational Biology  | 1553-7358  
80.  | Social Cognitive and Affective Neuroscience  | 1749-5016  
81.  | NeuroImage: Clinical  | 2213-1582  
82.  | Journal of Alzheimer's Disease  | 1875-8908  
83.  | Psychiatry and Clinical Neurosciences  | 1440-1819  
84.  | Wiley Interdisciplinary Reviews: Cognitive Science  | 1939-5086  
85.  | Neurobiology of Sleep and Circadian Rhythms  | 2451-9944  
86.  | Neurorehabilitation and Neural Repair  | 1545-9683  
87.  | Clinical Neurophysiology  | 1872-8952  
88.  | Investigative Ophthalmology and Visual Science  | 1552-5783  
89.  | Psychophysiology  | 1540-5958  
90.  | Journal of Experimental Psychology: General  | 1939-2222  
91.  | Journal of Neurodevelopmental Disorders  | 1866-1955  
92.  | Sleep Health  | 2352-7218  
93.  | Cognition  | 1873-7838  
94.  | Computational Intelligence and Neuroscience  | 1687-5273  
95.  | Journal of Experimental Psychology: Human Perception and Performance  | 1939-1277  
96.  | Neuromuscular Disorders  | 1873-2364  
97.  | Journal of Vestibular Research: Equilibrium and Orientation  | 1878-6464  
98.  | Brain and Language  | 1090-2155  
99.  | JARO - Journal of the Association for Research in Otolaryngology  | 1438-7573   
100.  | Topics in Cognitive Science  | 1756-8765   
101.  | Ophthalmic and Physiological Optics  | 1475-1313   
102.  | Behavioral Sleep Medicine  | 1540-2010   
103.  | Early Intervention in Psychiatry  | 1751-7893   
104.  | Current protocols in mouse biology  | 2161-2617   
105.  | Journal of Fluency Disorders  | 1873-801X   
106.  | Journal of Research on Adolescence  | 1532-7795   
107.  | International Journal of Behavioral Development  | 1464-0651   
108.  | Language, Cognition and Neuroscience  | 2327-3801   
109.  | Aphasiology  | 1464-5041   
110.  | Journal of Neurolinguistics  | 1873-8052   
111.  | Attention  Perception  and Psychophysics  | 1943-393X   
112.  | Journal of Language Evolution  | 2058-458X   
113.  | Phenomenology and the Cognitive Sciences  | 1568-7759   
114.  | Journal of Communication Disorders  | 1873-7994   
115.  | Mind, Culture, and Activity  | 1532-7884   

We then used Journal ISSN numbers to create a PubMed search string for the above 115 journals: 

1546-1726[Journal] OR 1471-0048[Journal] OR 1759-4766[Journal] OR 1873-1635[Journal] OR 1879-307X[Journal] OR 1097-4199[Journal] OR 2215-0374[Journal] OR 1545-4126[Journal] OR 1432-0533[Journal] OR 1552-5279[Journal] OR 1476-5578[Journal] OR 1532-2955[Journal] OR 1873-2402[Journal] OR 1873-5118[Journal] OR 1750-1326[Journal] OR 1878-108X[Journal] OR 1568-1637[Journal] OR 1460-2075[Journal] OR 1531-8249[Journal] OR 1873-7528[Journal] OR 1095-6808[Journal] OR 1531-8257[Journal] OR 2397-3374[Journal] OR 1420-9071[Journal] OR 1090-2139[Journal] OR 1879-0445[Journal] OR 1939-843X[Journal] OR 1873-6882[Journal] OR 1365-2990[Journal] OR 1528-1167[Journal] OR 1559-7016[Journal] OR 1545-7885[Journal] OR 1879-2782[Journal] OR 2047-9158[Journal] OR 1089-4098[Journal] OR 1098-1136[Journal] OR 2041-4889[Journal] OR 2050-084X[Journal] OR 1095-9572[Journal] OR 1875-6190[Journal] OR 2470-5470[Journal] OR 2373-8057[Journal] OR 1742-2094[Journal] OR 1876-4754[Journal] OR 1559-1182[Journal] OR 0925-2312[Journal] OR 2040-2392[Journal] OR 1477-0970[Journal] OR 1530-888X[Journal] OR 1758-9193[Journal] OR 1750-3639[Journal] OR 1749-6632[Journal] OR 1460-2199[Journal] OR 2352-2895[Journal] OR 1866-3389[Journal] OR 2451-9022[Journal] OR 1866-9964[Journal] OR 1095-953X[Journal] OR 1958-5969[Journal] OR 2158-3188[Journal] OR 1534-6293[Journal] OR 1097-0193[Journal] OR 8686-01X[Journal] OR 1754-8411[Journal] OR 2051-5960[Journal] OR 1528-8447[Journal] OR 1558-1497[Journal] OR 1467-7687[Journal] OR 1090-2430[Journal] OR 1932-846X[Journal] OR 1873-3360[Journal] OR 1558-0210[Journal] OR 1557-1904[Journal] OR 1399-5618[Journal] OR 1469-8749[Journal] OR 2472-1751[Journal] OR 1468-2079[Journal] OR 2352-1546[Journal] OR 1553-7358[Journal] OR 1749-5016[Journal] OR 2213-1582[Journal] OR 1875-8908[Journal] OR 1440-1819[Journal] OR 1939-5086[Journal] OR 2451-9944[Journal] OR 1545-9683[Journal] OR 1872-8952[Journal] OR 1552-5783[Journal] OR 1540-5958[Journal] OR 1939-2222[Journal] OR 1866-1955[Journal] OR 2352-7218[Journal] OR 1873-7838[Journal] OR 1687-5273[Journal] OR 1939-1277[Journal] OR 1873-2364[Journal] OR 1878-6464[Journal] OR 1090-2155[Journal] OR 1438-7573[Journal] OR 1756-8765[Journal] OR 1475-1313[Journal] OR 1540-2010[Journal] OR 1751-7893[Journal] OR 2161-2617[Journal] OR 1873-801X[Journal] OR 1532-7795[Journal] OR 1464-0651[Journal] OR 2327-3801[Journal] OR 1464-5041[Journal] OR 1873-8052[Journal] OR 1943-393X[Journal] OR 2058-458X[Journal] OR 1568-7759[Journal] OR 1873-7994[Journal] OR 1532-7884[Journal]  

In order to narrow down the search results to mata-analyses of non-human animal studies we applied the following PubMed filters: "Meta-Analysis", "Other Animals" (equivalen to "non-human animals"), publication date within the last 10 years. The PubMed search was performed in August 2021.   

We note that out of the 115 journal listed above three appeared to not be indexed in PubMed:  1939-843X (Journal of Abnormal Psychology), 8686-01X (Translational Stroke Research), 1873-8052 (Journal of Neurolinguistics). Thus, the search was effectively performed on the remaining 112 journals.   

## Inclusion criteria at title and abstract screening phase  
- Include studies claiming to be/contain meta-analysis (exclude protocols, methods, opinion articles, corrections, narrative reviews, etc.)  
- Include analyses of only animal data or mixed animal and human data (exclude exclusively human-focused studies)   
- Include studies with clear neurobiology/biomedical focus, such as experimental preclinical *in vivo* animal research (exclude studies ecological, evolutionary, agricultural or conservation focus).   
- Analysed outcomes should be related to neurobiology or behaviour, including meta-analyses of genomic or transcriptomic data, brain imaging.  

## Abstract screening procedure and results  
We used Rayyan QCRI software to screen 185 unique bibliographic records downloaded from Pubmed. two researchers (YY, ML) independently performed the screening assessing titles abstracts and keywords of each article. This screening resulted in 78 articles included for full-text re-assessment and data data extraction.  

## Inclusion criteria at full-text screening  
- Full-text studies should fulfill the same criteria as defined for the title and abstract screening phase  
AND  
- Studies to be included for detailed data extraction should be performed using formal meta-analytic methodology (i.e. meta-analytic or meta-regression models combining weighted effect sizes from multiple primary studies ("formal meta analysis")  

## Full text screening and data extraction  
Two researchers (YY, ML) independently performed full-text screening was performed concurrently with data extraction using a two-part custom Google Form (https://forms.gle/AYoU1BtkU4PgkmAy9). to pilot the form, randomly selected 14 papers were screened and extracted independently by to researchers (YY, ML). Any potential disagreements were resolved by discussion, and the form was refined as needed, before the main round of independent full-text screening and data extraction.  

The first part of the full-text screening and data extraction form was used to collect bibliographic information about each screened study, specifically: paper's title,	first author's family name,	publication journal name,	publication year, and article doi. The multiple-choice questions in the second part of the form are presented in **Table S1** below.:  

#### Table S2  
Question | Answer options  
---------| --------------------------  
Q1 Is this formal meta-analysis (weighted) of non-human animal studies? | Formal MA; Bioinformatic MA (e.g. transcriptional and genome-wide association studies); Image MA (e.g. fMRI); Self-claiming as containing a meta-analysis (or meta-regression) but actually no formal meta-analysis/regression was conducted; unclear/other (add comment)  
Q2 What is the type of effect sizes used?	| SMD (d, g, etc.); raw mean; correlation (r or Zr); Response Ratio (lnRR); variance (VR or lnCVR); 2 by 2 binary data (RR, OR); unclear/other (add comment)  
Q3 What are the types of statistical models used for MA? | fixed-effect; random-effects; meta-regression (weighted linear-regression); multilevel; multivariate; unclear/other (add comment)  
Q4 Does the paper account for heterogeneity? | not claiming to account for or explore heterogeneity; accounted by meta-regression, subgroup analysis or stratified MA; unclear/other (add comment)  
Q5 What are the types of heterogeneity indices reported? | no heterogeneity indices reported; tau^2; Q; H; I^2; Prediction interval; unclear/other (add comment)  
Q6 Does the study report N (the number of included studies) and k (the number of effect sizes)? | Both reported (add values to the comment); N reported (add values to the comment); k reported (add values to the comment); None reported (add comment)  
Q7 N | Enter included papers number (N) in the larges statistical analysis/model. If not reported, enter "NA"  
Q8 k | Enter included effect sizes number (k) in the larges statistical analysis/model. If not reported, enter "NA"  
Q9 Does the study has the sign of data non-independence? | k/N=1 indicates independent effect sizes; k/N>1 indicates non-independent effect sizes; unclear/other (add comment)   
Q10 Does the study acknowledge the issue of data non-independence? | Acknowledges effect sizes are not fully independent; Claiming effect sizes are independent; unclear/other (add comment)  
Q11 Does the paper attempt to address any issues related to data non-independence? | paper explicitly models non-independence (e.g. using multilevel, multivariate models); paper deals with non-independence in other than modelling way (e.g. aggregation, sampling); ignoring or only partially dealing with non-independence (using fixed- or random-effects models for dependent effects); N/A (it is claimed all data are independent); unclear/other (add comment)  
Q12 Does the paper report any approach to control for the sampling error non-independence (e.g. VCV matrix, sample size adjustment)? | VCV matrix; sample size adjustment; cluster-robust standard errors (sandwich robust variance estimation); N/A (it is claimed all data are independent); not controlling for the sampling error non-independence in any way; unclear/other (add comment)  
Q13 What approaches were used to testing for publication bias | Not claiming to do tests for publication bias; Funnel plots; Regression-based methods (e.g., Egger's test); Fail-safe N; Trim-and-fill tests; Correlation-based methods (e.g., rank); P-value-based methods; Selection models; Normal quantile plot (Q-Q); Time-lag bias tests unclear/other (add comment)  
Q14 What software was used for analyses? | RevMan; Stata packages; R packages; MetaWin; Comprehensive Meta-Analysis; unclear/other (add comment)  

Each question was followed by a dedicated comment field used to record any additional details, including relevant quotes from the article. Any papers that were coded as "no" at question Q1 (full-text screening question - whether the paper includes a formal meta-analysis of non-human animal data) were excluded, i.e. not subject to further data extraction and analyses.  


## Screening results  
Out of the 78 full-text articles screened, 16 were deemed ineligible for data extraction (**Table S2**). The final dataset (after filtering out ineligible papers) comprises 62 formnal meta-analyses of non-human animal neurobiology-related questions. The data extraction spreadsheet is saved as *MA_Neurobiology_Survey_final_clean.xlsx*.  Below, we present a summary of the extracted data.

```{r load data}
rawdata <- read_excel(here("data", "MA_Neurobiology_Survey_final_clean.xlsx"), sheet = 1)
#dim(rawdata) #78 papers 33 columns
#names(rawdata)
#glimpse(rawdata) # taking a quick look
```

### Summary tables 

#### Table S4
List of articles excluded at full-text screening, with reasons for exclusion.     
```{r data table excluded articles}
#table(rawdata$"exclusion_reason") #table of exclusion resons for the 4 studies
#remove 46 included studies and select a few relevant columns
rawdata_excl <- rawdata %>% filter_at(6, all_vars(.!="Formal MA")) %>% select(c(2,4,3,1,5,6)) 
#dim(rawdata_excl) #16 articles, 6 columns
#names(rawdata_excl)
names(rawdata_excl) <- c("First_author", "Year", "Journal", "Title", "DOI", "Exclusion_reason")
#make a table of excluded  studies
t2 <- rawdata_excl %>%   DT::datatable(rownames = FALSE, width = "100%", options = list(dom = 't', scrollY = '700px', pageLength=20)) #, pageLength = 2
t2
```

#### Table S3
List of included articles.   
```{r data table included articles}
#remove 4 excluded studies and remove all columns with "Comment", "checked" and then first 2 columns
rawdata_incl <- rawdata %>% filter_at(6, all_vars(.=="Formal MA")) %>% select(-starts_with("Comment ")) 
#make a table of included studies
rawdata_incl2 <- select(rawdata_incl, c(2,4,3,1,5)) 
names(rawdata_incl2) <- c("First_author", "Year", "Journal", "Title", "DOI")
t3 <- rawdata_incl2 %>% DT::datatable(rownames = FALSE, width = "100%", options = list(dom = 't', scrollY = '700px', pageLength=20)) 
t3
```

### Preprocessing extracted data
Data cleaning before plotting.      

```{r clean data}
#replace column names with shorter variable names for rawdata_incl analyses
names(rawdata_incl)  <- c("Title",
                 "Author",
                 "Journal",
                 "Year",
                 "DOI",
                 "Meta_analysis",
                 "Effect_size",
                 "Statistical_models",
                 "Heterogeneity_accounted",
                 "Heterogeneity_indices",
                 "Reporting_N_k",
                 "N",
                 "k",
                 "Nonindependent_data",
                 "Acknowledging_data_nonindependence",
                 "Addressing_data_nonindependence",
                 "Sampling_error_nonindependence",
                 "testing_publication_bias",
                 "Software_used")
```


### Plots

#### Publication year

```{r plot year, fig.dim = c(8, 2)}
count(rawdata_incl, Year) %>%
  mutate(class=factor(Year, levels=Year)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
#  geom_text(aes(label=scales::comma(n)), hjust=0, nudge_y=1) +
#  coord_flip() +
  scale_y_continuous(breaks=seq(0,20,1)) +
  labs(x="", y="Article count", title="When it was published?")
# ggsave(filename = "./figures/FigS01.pdf",
#        plot = plot_qual,
#        height = 5, width = 10,
#        device = cairo_pdf)
```

#### Publication journal

```{r plot journal, fig.dim = c(8, 6)}
count(rawdata_incl, Journal) %>%
  arrange(n) %>%
  mutate(class=factor(Journal, levels=Journal)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
 # geom_text(aes(label=scales::comma(n)), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,20,5)) +
  labs(x="", y="Article count", title="Where it was published?")
# ggsave(filename = "./figures/FigS02.pdf",
#        plot = plot_qual,
#        height = 12, width = 10,
#        device = cairo_pdf)
```

#### Effect size type

```{r plot effect size, fig.dim = c(8, 4)}
#table(rawdata_incl$Effect_size) 
Effect_size_sep <- separate_rows(rawdata_incl, Effect_size, sep = ";") #split rows with multiple values
Effect_size_sep$Effect_size <- as.factor(Effect_size_sep$Effect_size)
levels(Effect_size_sep$Effect_size) <- c("2 by 2 binary data (RR, OR)",
                                         "correlation (r or Zr)",
                                         "Normalised mean difference",
                                         "Response Ratio (lnRR)",
                                         "SMD (d, g, etc.)",
                                         "unclear/other",
                                         "variance (VR or lnCVR)" )
### Count the relative frequency of factor levels
Effect_size_sep %>%
  group_by(Effect_size) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))


count(Effect_size_sep, Effect_size) %>%
  arrange(n) %>%
  mutate(class=factor(Effect_size, levels=Effect_size)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,50,5)) +
  labs(x="", y="Article count", title="What type of effect size was used?", caption="Note: some studies used more than one")
# ggsave(filename = "./figures/FigS03.pdf",
#        plot = plot_qual,
#        height = 5, width = 10,
#        device = cairo_pdf)
```


#### Statistical models

```{r plot statistical models, fig.dim = c(8, 5)}
#table(rawdata_incl$Statistical_models)
Statistical_models_sep <- separate_rows(rawdata_incl, Statistical_models, sep = ";") #split rows with multiple values
Statistical_models_sep$Statistical_models <- as.factor(Statistical_models_sep$Statistical_models)
levels(Statistical_models_sep$Statistical_models) <- c("fixed-effect",
                                                       "meta-regression", 
                                                       "multilevel",
                                                       "random-effects", 
                                                       "unclear/other" )

### Count the relative frequency of factor levels
Statistical_models_sep %>%
  group_by(Statistical_models) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))


count(Statistical_models_sep, Statistical_models) %>%
  arrange(n) %>%
  mutate(class=factor(Statistical_models, levels=Statistical_models)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,50,5)) +
  labs(x="", y="Article count", title="What type of statistical model was used?", caption="Note: some studies used more than one")
# ggsave(filename = "./figures/FigS04.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Heterogeneity accounted

```{r plot heterogeneity accounted, fig.dim = c(8, 5)}
#table(rawdata_incl$Heterogeneity_accounted)
rawdata_incl$Heterogeneity_accounted <- as.factor(rawdata_incl$Heterogeneity_accounted)
levels(rawdata_incl$Heterogeneity_accounted) <- c("meta-regression, subgroup\nanalysis or stratified MA",
                                                  "not claiming to account for\nor explore heterogeneity",
                                                  "unclear/other")

### Count the relative frequency of factor levels
rawdata_incl %>%
  group_by(Heterogeneity_accounted) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))

count(rawdata_incl, Heterogeneity_accounted) %>%
  arrange(n) %>%
  mutate(class=factor(Heterogeneity_accounted, levels=Heterogeneity_accounted)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,60,5)) +
  labs(x="", y="Article count", title="How heterogeneity was accounted for or explored?")
# ggsave(filename = "./figures/FigS05.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Heterogeneity indices

```{r plot heterogeneity indices, fig.dim = c(8, 5)}
#table(rawdata_incl$Heterogeneity_indices) 
Heterogeneity_indices_sep <- separate_rows(rawdata_incl, Heterogeneity_indices, sep = ";") #split rows with multiple values
Heterogeneity_indices_sep$Heterogeneity_indices <- as.factor(Heterogeneity_indices_sep$Heterogeneity_indices)
levels(Heterogeneity_indices_sep$Heterogeneity_indices) <- c("H",
                                                             "I^2",
                                                             "no heterogeneity\nindices reported",
                                                             "Prediction interval",
                                                             "Q",
                                                             "tau^2",
                                                             "unclear/other")


### Count the relative frequency of factor levels
Heterogeneity_indices_sep %>%
  group_by(Heterogeneity_indices) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))


count(Heterogeneity_indices_sep, Heterogeneity_indices) %>%
  arrange(n) %>%
  mutate(class=factor(Heterogeneity_indices, levels=Heterogeneity_indices)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,50,5)) +
  labs(x="", y="Article count", title="How heterogeneity was quantified?", caption="Note: some studies used more than one")
# ggsave(filename = "./figures/FigS06.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```

Note: another index that was commonly mentioned was Chi^2.   


#### Reporting N and k

```{r plot reporting N and k, fig.dim = c(8, 5)}
#table(rawdata_incl$Reporting_N_k)
rawdata_incl$Reporting_N_k <- as.factor(rawdata_incl$Reporting_N_k)
levels(rawdata_incl$Reporting_N_k) <- c("N and k reported",
                                        "k reported",
                                        "N reported",
                                        "None reported")
count(rawdata_incl, Reporting_N_k) %>%
  arrange(n) %>%
  mutate(class=factor(Reporting_N_k, levels=Reporting_N_k)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,60,5)) +
  labs(x="", y="Article count", title="Were used N (number of articles) and k (number of effect sizes) reported?")
# ggsave(filename = "./figures/FigS07.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Reported N

```{r reported N, fig.dim = c(8, 5)}
#table(rawdata_incl$N)
 
#calculate the percentage of N>50
nrow(rawdata_incl[rawdata_incl$N>40, ])/nrow(rawdata_incl)
 
ggplot(data=subset(rawdata_incl, !is.na(N)) , aes(x=as.numeric(N))) + 
  geom_histogram(binwidth=10) +
  labs(x="", y="Article count", title="What was N (number of analysed articles)?")
# ggsave(filename = "./figures/FigS08.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Reported k

```{r reported k, fig.dim = c(8, 5)}
#table(rawdata_incl$k)
k_data <- rawdata_incl  %>% filter(k != "NA") %>% filter(k != "2300000") %>% mutate(k = as.integer(k))

#calculate the percentage of k>50
nrow(k_data[k_data$k>50, ])/nrow(k_data)

ggplot(data=k_data , aes(k)) + 
  geom_histogram(binwidth=10) +
  labs(x="", y="Article count", title="What was k (number of analysed effect sizes)?", caption="Note: one study with k>1000000 not shown")
# ggsave(filename = "./figures/FigS09.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Nonindependent data based on k/N ratio

```{r plot Nonindependent_data, fig.dim = c(8, 3)}
#table(rawdata_incl$Nonindependent_data)
rawdata_incl$Nonindependent_data <- as.factor(rawdata_incl$Nonindependent_data)
levels(rawdata_incl$Nonindependent_data) <- c("k/N=1 indicates\nindependent effect sizes", 
                                              "k/N>1 indicates\nnon-independent effect sizes",
                                              "unclear/other")


### Count the relative frequency of factor levels
rawdata_incl %>%
  group_by(Nonindependent_data) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))

count(rawdata_incl, Nonindependent_data) %>%
  arrange(n) %>%
  mutate(class=factor(Nonindependent_data, levels=Nonindependent_data)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,60,5)) +
  labs(x="", y="Article count", title="Does k/N ratio indicate data non-independence?") #, caption="Literature survey result"
# ggsave(filename = "./figures/FigS10.pdf",
#        plot = plot_qual,
#        height = 5, width = 10,
#        device = cairo_pdf)
```


#### Acknowledging non-independence of effect sizes 

```{r plot acknowledging non-independence , fig.dim = c(8, 3.5)}
#table(rawdata_incl$Acknowledging_data_nonindependence)
rawdata_incl$Acknowledging_data_nonindependence <- as.factor(rawdata_incl$Acknowledging_data_nonindependence)
levels(rawdata_incl$Acknowledging_data_nonindependence) <- c("Acknowledges effect sizes\nare not fully independent",
                                                    "Claiming effect sizes\nare independent",
                                                    "unclear/other",
                                                    "unclear/other")
count(rawdata_incl, Acknowledging_data_nonindependence) %>%
  arrange(n) %>%
  mutate(class=factor(Acknowledging_data_nonindependence, levels=Acknowledging_data_nonindependence)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,40,5)) +
  labs(x="", y="Article count", title=" Did the authors acknowledge non-independence of effect sizes?") #, caption="Literature survey result"
# ggsave(filename = "./figures/FigS11.pdf",
#        plot = plot_qual,
#        height = 6, width = 10,
#        device = cairo_pdf)
```


#### Addressing data non-independence  

```{r plot adressing non-independence, fig.dim = c(8, 3.5)}
#table(rawdata_incl$Addressing_data_nonindependence)
rawdata_incl$Addressing_data_nonindependence <- as.factor(rawdata_incl$Addressing_data_nonindependence)
levels(rawdata_incl$Addressing_data_nonindependence) <- c("ignoring or only partially\nadressing via modeling",
                                                          "N/A (it is claimed\nall data are independent)",
                                                          "adressing in other than\nmodeling way",
                                                          "explicitly modeling\nnon-independence",
                                                          "unclear/other")

### Count the relative frequency of factor levels
rawdata_incl %>%
  group_by(Addressing_data_nonindependence) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))

count(rawdata_incl, Addressing_data_nonindependence) %>%
  arrange(n) %>%
  mutate(class=factor(Addressing_data_nonindependence, levels=Addressing_data_nonindependence)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,40,5)) +
  labs(x="", y="Article count", title=" How did the authors address non-independence of effect sizes?") #, caption="Literature survey result"
# ggsave(filename = "./figures/FigS12.pdf",
#        plot = plot_qual,
#        height = 6, width = 10,
#        device = cairo_pdf)
```


#### Addressing sampling error non-independence  

```{r plot adressing SE non-independence, fig.dim = c(8, 3.5)}
#table(rawdata_incl$Sampling_error_nonindependence)
rawdata_incl$Sampling_error_nonindependence <- as.factor(rawdata_incl$Sampling_error_nonindependence)
levels(rawdata_incl$Sampling_error_nonindependence) <- c("using cluster-robust\nstandard errors",
                                                          "N/A (it is claimed\nall data are independent)",
                                                          "not adressing in any way",
                                                          "using sample size\nadjustment",
                                                          "unclear/other",
                                                          "using VCV matrix")

### Count the relative frequency of factor levels
rawdata_incl %>%
  group_by(Sampling_error_nonindependence) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))

count(rawdata_incl, Sampling_error_nonindependence) %>%
  arrange(n) %>%
  mutate(class=factor(Sampling_error_nonindependence, levels=Sampling_error_nonindependence)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,40,5)) +
  labs(x="", y="Article count", title=" How did the authors address non-independence of sampling errors?") #, caption="Literature survey result"
# ggsave(filename = "./figures/FigS13.pdf",
#        plot = plot_qual,
#        height = 6, width = 10,
#        device = cairo_pdf)
```


#### testing for publication bias

```{r plot publication bias, fig.dim = c(8, 5)}
#table(rawdata_incl$testing_publication_bias)
testing_publication_bias_sep <- separate_rows(rawdata_incl, testing_publication_bias, sep = ";") #split rows with multiple values
testing_publication_bias_sep$testing_publication_bias <- as.factor(testing_publication_bias_sep$testing_publication_bias)
levels(testing_publication_bias_sep$testing_publication_bias) <- c("Correlation-based methods",
                                                                   "Fail-safe N",
                                                                   "Funnel plots",
                                                                   "Not testing",
                                                                   "Regression-based methods",
                                                                   "Time-lag bias test",
                                                                   "Trim-and-fill test",
                                                                   "unclear/other")

testing_publication_bias_sep %>%
  group_by(testing_publication_bias) %>%
  summarise(n = n()) %>%
  mutate(Freq = round(100*n/sum(n), 0))


count(testing_publication_bias_sep, testing_publication_bias) %>%
  arrange(n) %>%
  mutate(class=factor(testing_publication_bias, levels=testing_publication_bias)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,50,5)) +
  labs(x="", y="Article count", title="Which publication bias tests were used?", caption="Note: some studies used more than one")
# ggsave(filename = "./figures/FigS14.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```


#### Software 

```{r plot software, fig.dim = c(8, 4)}
#table(rawdata_incl$Software_used)
Software_used_sep <- separate_rows(rawdata_incl, Software_used, sep = ";") #split rows with multiple values
Software_used_sep$Software_used <- as.factor(Software_used_sep$Software_used)
levels(Software_used_sep$Software_used) <- c("Comprehensive\nMeta-Analysis", 
                                             "R packages",
                                             "RevMan",
                                             "Stata packages",
                                             "unclear/other")
count(Software_used_sep, Software_used) %>%
  arrange(n) %>%
  mutate(class=factor(Software_used, levels=Software_used)) %>%
  ggplot(aes(x = class, y = n)) +
  geom_bar(stat = "identity", position = "dodge") +
  geom_text(aes(label=as.integer(scales::comma(n))), hjust=0, nudge_y=1) +
  coord_flip() +
  scale_y_continuous(breaks=seq(0,30,5)) +
  labs(x="", y="Article count", title="What type of statistical software was used?", caption="Note: some studies used more than one")
# ggsave(filename = "./figures/FigS15.pdf",
#        plot = plot_qual,
#        height = 8, width = 10,
#        device = cairo_pdf)
```

Retrieving additional info on other software and packages used from the comments field.   

```{r other software packages used}
#names(rawdata) collect information on packages and other software used  
software <- rawdata$"Comment for Q14" %>% na.omit()
software <- software[! software %in% c("no reporting", "not mentioned", "not reported", "not formal MA", "no MA of animal studies", "no formal MA of animal studies", "no formal MA", "no formal MA; no info", "no detailed packagesmentioned")] %>% tolower()
software2 <- unlist(strsplit(software, ";" )) #split strings at ; 
software2 <- trimws(software2, whitespace = "[\\h\\v]") #remove leading and trialing whitespace
table(software2)  
#count use of metafor and meta R packages
#sum(software2=="metafor")
#sum(software2=="meta")
```

R package *metafor* was the most used, with `r sum(software2=="metafor")` mentions. R package *meta came second, with `r sum(software2=="meta")` mentions.   


## TODO
- select plots for presentation in the main text
- make a multi-panel figure for the main text